pages/1_orders_and_revenue.py

```diff
@@ -18,49 +18,163 @@
 
 st.divider()
 
-st.dataframe(data.head(10)) ## This is here just as an example. This can be deleted during development and before release.
+## Ensure all date columns are in datetime format
+date_columns = ['created_at', 'customer_created_at', 'payment_at', 'subscription_period_started_at', 'subscription_period_ended_at']
+for col in date_columns:
+    if col in data.columns:
+        data[col] = pd.to_datetime(data[col], errors='coerce')
 
 st.divider()
 
-# Everything below this point will be generated by the assigned analyst. 
-# The framework has been set, but no visualization or data processing has been applied.
-# Please perform any data processing in this file and not within the filter files. We can discuss upon completion if it makes sense to add any code to the filters file.
-
-## KPI Metrics which need to be updated.
+# KPI Metrics
 with st.container():
     col1, col2, col3, col4 = st.columns(4)
 
+    total_revenue = data['total_amount'].sum()
+    number_of_orders = data['header_id'].nunique()
+    number_of_customers = data['customer_id'].nunique()
+    new_customers = data[data['customer_created_at'] == data['created_at']]['customer_id'].nunique()
```

Comment:
This code will always always show new customers of 0 since it is only counting a new customer if they were created on the exact same day as the order being processed.

Instead I would recommend the following changes. This ensures we are counting a new customer in the period. If the customer created date is between the min and max created_at timestamps, then we can count them as a new customer in the period.
```suggestion
    min_created_at = data['created_at'].min()
    max_created_at = data['created_at'].max()
    new_customers = data[(data['customer_created_at'] >= min_created_at) & (data['customer_created_at'] <= max_created_at)].shape[0]
```

Reply:
Ah good catch!